server/src/main/java/org/kaazing/gateway/server/context/resolve/DefaultServiceContext.java

```diff
@@ -729,7 +729,28 @@ private void injectServiceOptions(String transportURI, Map<String, Object> optio
                                 getProperties().get("encryption.key.alias"));
                         options.put(format(optionPattern, SERVICE_DOMAIN),
                                 getProperties().get("service.domain"));
+                    }
+                }
 
+                // TCP
+                for (String optionPattern : asList("tcp.%s")) {
+                    // NO REALM_NAME as this will be an accept/connect option
+                    String tcpRealmOptionName = format(optionPattern, REALM_NAME);
+                    String tcpRealmName = (String) options.get(tcpRealmOptionName);
+                    if (tcpRealmName != null) {
+                        // check if it's the same as the configured realm
+                        if (!serviceRealmContext.getName().equals(tcpRealmName)) {
+                            logger.error("{} configuration error: {} needs to be set to the same value as the configured realm",
+                                    serviceName, tcpRealmOptionName);
+                            throw new IllegalArgumentException(
+                                    tcpRealmOptionName + " needs to be the same as the configured realm");
+                        }
+                        options.put(format(optionPattern, LOGIN_CONTEXT_FACTORY), serviceRealmContext.getLoginContextFactory());
+//                    initial support will just be authenticate or not
+//                    options.put(format(optionPattern, REQUIRED_ROLES),
+//                            getRequireRoles());
+//                    options.put(format(optionPattern, REALM_USER_PRINCIPAL_CLASSES.name()),
+//                            getUserPrincipalClasses(serviceRealmContext.getUserPrincipalClasses()));
```

Comment:
What is the rationale for requiring the `tcp.realm` to match the `<service><realm-name>`?

For example, could the above code (lines 735-753) be replaced by the following:

```
UnaryOperator<String> tcpFormat = o -> String.format("tcp.%s", o);
String tcpRealm = (String) options.get(tcpFormat.apply("realm"));
if (tcpRealm != null) {
    options.put(tcpFormat.apply(LOGIN_CONTEXT_FACTORY), serviceRealmContext.getLoginContextFactory());
}
```